.travis.yml

```diff
@@ -1,130 +1,166 @@
+    #     - python -m pip install --upgrade pip
+    #   env: PATH=/c/Python310:/c/Python310/Scripts:$PATH
+    #   install:
+    #     # Install Micromamba
+    #     - Invoke-WebRequest -Uri 'https://micro.mamba.pm/api/micromamba/win-64/1.5.8' -OutFile 'micromamba.tar.bz2'
+    #     - Expand-Archive -Path 'micromamba.tar.bz2' -DestinationPath 'micromamba'
+    #     - Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope CurrentUser
+    #     - micromamba shell init -s powershell -p ~/micromamba
+    #     - micromamba activate base
+    #     # Update python version in environment
+    #     - (Get-Content -Path 'environment.yml') -replace '(python=)(.*)', ('$1' + $env:TRAVIS_PYTHON_VERSION) | Set-Content -Path 'environment.yml'
+    #     # Create conda environment
+    #     - micromamba install xtensor -c conda-forge -y
+    #     # move requirements.txt to the /tmp folder, required so that .environment.yml finds it
+    #     - Copy-Item -Path 'requirements.txt' -Destination 'C:\tmp\requirements.txt'
+    #     # Create the environment:
+    #     - micromamba create -f 'environment.yml' -y
+    #     # Activate and install test-only dependencies
+    #     - micromamba activate radis-env
+    #     - micromamba list
+    #     - pip install pytest-cov
+    #     - pip install -e . -v
+  ### Extra Python versions - start
+    - stage: Py312
+      name: "Python 3.12"
+      python: 3.12
+  ### Extra Python versions - end
     - stage: lint
       name: Code Quality Checks
+      python: 3.10
       install:
         - pip install pre-commit
         - pre-commit install
+      before_script: skip #do NOT run the before_script of "tests" stage
       script:
         - pre-commit run --all-files
       after_failure: #better for checking the origin of a bug
         - cat /home/travis/.cache/pre-commit/pre-commit.log
 
+    - stage: build_wheels
+      name: Build the Wheels on Linux, Windows
+      # Added from https://github.com/pypa/cibuildwheel/blob/main/examples/travis-ci-deploy.yml; see https://github.com/radis/radis/issues/650
+      include:
+        # perform a linux build
+        - services: docker
+        # and a windows build
+        - os: windows
+          language: shell
+          before_install:
+            - choco upgrade python -y --version 3.8.6
+            - export PATH="/c/Python38:/c/Python38/Scripts:$PATH"
+            # make sure it's on PATH as 'python3'
+            - ln -s /c/Python38/python.exe /c/Python38/python3.exe
+
+      install: skip
+        #- python3 -m pip install cibuildwheel==2.19.1 #If you intend to build a pure-Python wheel, you don't need cibuildwheel
+
+      before_script: skip #do NOT run the before_script of "tests" stage
+
+      script:
+        # build the wheels, put them into './dist'
+        # - python3 -m cibuildwheel --output-dir dist #If you intend to build a pure-Python wheel, you don't need cibuildwheel
+        - python3 -m pip wheel -w dist .
+
```

Comment:
@minouHub I don't think we need this (nor the "skip_cleanup: true") step since we're not deploying cibuildwheels.  Pure Python wheels are already built and deployed by the "deploy" stage ([distributions: "sdist bdist_wheel"](https://github.com/radis/radis/pull/656/files#diff-6ac3f79fc25d95cd1e3d51da53a4b21b939437392578a35ae8cd6d5366ca5485R176))

And unless I'm wrong, Pure Python wheels are the same for every OS (Linux/Windows)

Reply:
@erwanp, I just applied what is recommended by Travis https://docs.travis-ci.com/user/job-lifecycle/#deploying-your-code
To be honest, I'm completely new to this and simply applying what I read

> And unless I'm wrong, Pure Python wheels are the same for every OS (Linux/Windows)

You're right, but I don't think that this is comment is related to `skip_cleanup: true`

Reply:
For deployment; I think it is handled by : 

https://docs.travis-ci.com/user/deployment/pypi/#uploading-different-distributions

```
deploy:
  provider: pypi
  username: ...
  password: ...
  distributions: "sdist bdist_wheel" # Your distributions here
# If you specify bdist_wheel in the distributions, the wheel package will automatically be installed.
```

So I assume wheel packages are automatically created by the "deploy" stage too. 
- Unfortunately the Logs of the last push to Master branch [are not available anymore](https://app.travis-ci.com/github/radis/radis/jobs/552887026) so we don't know if wheels were built there.
- But, [Radis 0.14 deployed on PyPi](https://pypi.org/project/radis/#files) has a built wheel.  

So I'm confident an extra build_stage is not needed

Reply:
I just checked the travis.yaml file pushed on Nov 8, 2022 ([here](https://github.com/radis/radis/pull/532/files#)) and the 0.14 version released on Pypi on Nov 8, 2022.

I agree with you. The Python wheels are **probably** generated automatically. I'll switch back to the original version of the deploy phase and let you merge if you're happy